src/python/grpcio/grpc/__init__.py

```diff
@@ -833,7 +833,11 @@ class HandlerCallDetails(six.with_metaclass(abc.ABCMeta)):
 
 
 class GenericRpcHandler(six.with_metaclass(abc.ABCMeta)):
```

Comment:
The issue is in avoiding the situation of an application creating a `ServiceRpcHandler`, registering it with a server using `add_generic_rpc_handlers`, and then the server having to do dynamic type inspection and special casing along the lines of "well, which of the handlers just registered are `ServiceRpcHandler`s and which are merely `GenericRpcHandlers`?" when the application had that information on its side of the API. It's been really nice that we've been able to avoid dynamic type inspection so far and I'd like to keep it up.

With that pushing in the direction of a `Server.add_service_rpc_handlers` method the question of the design of the `ServiceRpcHandler` type is raised. How should it be called to service an RPC? By full-size `/my_package.my_subpackage.MyService/MyMethod` string? Or just by `MyMethod` string? Should `ServiceRpcHandler`s be free to refuse service of an RPC?

I think these questions can be answered in a way that allows `ServiceRpcHandler` to inherit the `service` method from `GenericRpcHandler`. Do you foresee any problems?